the-list-be/server.js

```diff
@@ -10,24 +10,51 @@ const uri = 'mongodb://localhost:27017';
 const client = new MongoClient(uri);
 const db = client.db('database');
 const students = db.collection('students');
+const dailyAttendance = db.collection('dailyAttendance');
 
-router.post('/students', async ctx => {
-  const insertResult = await students.insertOne(ctx.request.body);
-  ctx.body = insertResult;
+router.put('/attendance/:date', async ctx => {
+  await dailyAttendance.updateOne(
+    { date: ctx.params.date },
+    { $addToSet: { students: ctx.request.body._id } },
+    { upsert: true }
+  );
+  const check = await dailyAttendance.find({date: ctx.params.date}).toArray();
+  ctx.body = check;
+});
+
+router.get('/attendance/:date', async ctx => {
+  const FindStudentsInDate = await dailyAttendance.find({date: ctx.params.date}).toArray();
```

Comment:
Т.к. мы знаем, что запись с посещаемостью для каждого дня в БД одна, мы можем  делать `findOne` и убрать `toArray`.

```suggestion
  const FindStudentsInDate = await dailyAttendance.findOne({date: ctx.params.date});
```

Reply:
А чем `findOne` отличается от `find`?

Reply:
@Roha990, `findOne` возвращает первый документ, соответствующий запросу, `find` — возвращает все. Если документов, соответствующих запросу, в БД несколько или если нет уникального индекса на поле, по которому мы ищем, `findOne` будет работать быстрее, т.к. перестанет перебирать БД после того, как найдёт первый подходящий документ. В общем случае, когда нам нужен только один документ, лучше использовать `findOne`.